pkg/storage/mockstorage/store.go

```diff
@@ -0,0 +1,574 @@
+// Code generated by moq; DO NOT EDIT.
```

Comment:
i typically aver from checking generated code into version control, although i acknowledge that there's not a strict, dogmatic rule about this.  the thing is, i'm fairly scatterbrained:  if it's possible to forget to regenerate, i will eventually forget to regenerate.  so particularly, if CI could be responsible for keeping mocks in sync with the single source of truth, it would be much more convenient for me.

could we consider checking in `mockstorage/.gitkeep` only and changing `.circleci/config.yaml` to produce `mockstorage/store.go` on the fly?

meanwhile, for the use-case of running tests locally, we have a `Makefile` with a `test` target.  there could be a `generate` prereq for that target to generate mocks according to a `go:generate` comment like in the [suggested `moq` usage](https://github.com/matryer/moq#usage)?

yet, checking in the generated source could also be a considered and deliberate decision with some advantages that i'm not noticing and we can sync up via slack, zoom, or otherwise to get into the nitty-gritty of the tradeoffs involved.

Reply:
Yeah just wanted to avoid hacking through existing code with go generate stuff depending on your local env

Reply:
Yeah just wanted to avoid hacking through existing code with go generate stuff depending on your local env

Reply:
> if `// go:generate` is better than `hack/generate-xxx.sh`

no strong opinion here.  i was only repeating the documentation, and this is not a strong opinion compared to experience with actual usage.

Reply:
* [x] removed mock from git
* [x] chose go generate / removed hack script
* [x] added target in Makefile
* [x] included that in CI